src/main/scala/nodes/learning/KernelBlockModel.scala

```diff
@@ -0,0 +1,74 @@
+package nodes.learning
+
+import breeze.linalg._
+import edu.berkeley.cs.amplab.mlmatrix.{RowPartition, NormalEquations, BlockCoordinateDescent, RowPartitionedMatrix}
+import nodes.stats.{StandardScalerModel, StandardScaler}
+import nodes.util.{VectorSplitter, Identity}
+import org.apache.spark.broadcast.Broadcast
+import org.apache.spark.rdd.RDD
+import utils.{MatrixUtils, Stats}
+import workflow.{Transformer, LabelEstimator}
+
+
+/**
+ * @param models different kernel models for different lambdas
+ * @param blockSize blockSize to split data before applying transformations
+ * @param lambdas  lambdas corresponding to the models
+ * @param in the training Data
+ * @param kernelGen the kernel generator
+ * @param nTrain number j training examples
+ **/
+
+class KernelBlockModel(
+  model: DenseMatrix[Double],
+  blockSize: Int,
+  lambdas: Array[Double],
+  kernelGen: KernelGenerator,
```

Comment:
So having made the kernelGenerator a transformer we shouldn't need to take this in as a part of the model ? Shouldn't it already be a part of the pipeline ?